### ISSUE TITLE
Would it make sense to move SegmentationSmoothing to Slicer Core?

### ISSUE THREAD

**Issue by che85** [SlicerProstate/SlicerProstate#49]
@fedorov Slicetracker still has a dependency to SlicerProstate because of the SegmentationSmoothing which is needed for the result of DeepInfer. 

Since there is the Segment Editor, would it make sense to move this to Slicer Core? It's more of a general purpose module than prostate specific...

**Comment by lassoan** [comment]
There are 4 different smoothing methods implemented in the Segment Editor in Slicer core. Would you be able to just use those? See examples here: https://www.slicer.org/wiki/Documentation/Nightly/ScriptRepository#How_to_run_segment_editor_effects_from_a_script.

**Comment by che85** [comment]
@lassoan Good to know. Let me check. Thanks

**Comment by che85** [comment]
@lassoan The thing is that DeepInfer returns a vtkMRMLLabelMapNode. So it would make it even more complicated I guess, right?

**Comment by lassoan** [comment]
If you need segment-wise Gaussian smoothing then it is not even worth creating a CLI for that. You can just run through all the segments using a SimpleITK filter. If you need more sophisticated smoothing, for example joint smoothing of non-overlapping segments with watertight interfaces then it is worth to create a segmentation, import a labelmap (that's a single line of code, won't complicate things), and execute smoothing effect with joint smoothing option.

**Comment by fedorov** [comment]
@che85 I agree with Andras, we should see if we can just use the Segment Editor smoothing effect.

@lassoan is this the source code: http://apidocs.slicer.org/master/SegmentEditorSmoothingEffect_8py_source.html? The link you mentioned does not explain the smoothing effect options.

One difference is that the CLI Christian mentioned first does isotropic resampling of the label, and then applies Gaussian smoothing. Christian, can you take a representative dataset and compare the results? If the CLI option gives better results than the effect, we can re-implement it using SimpleITK. I don't remember now why I did it in C++ in the first place!

**Comment by lassoan** [comment]
Resampling to an isotropic space should the first step of any segmentation. You can crop&resample the original grayscale volume (the advantage is that usually the volume will not get much larger, as the size increase due to isotropic resampling is compensated by cropping); or set the Segmentation node's reference geometry to use isotropic, potentially higher resolution grid (this option is not available from the GUI in a convenient way, so you have to set the reference geometry conversion parameter programmatically).

Description of the different options are in the "helpText". Let me know if you need more details on any of the options.
